apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/utils/TradingViewUtils.java

```diff
@@ -0,0 +1,185 @@
+            result.volumeto = BigDecimal.ZERO;     
+        }
+        return result;         
+    }        
+
+
+        
+    static public TradingDataOutput getDataForIntervalFromOffers( String fsym, String tsym, Integer toTs, Integer limit, Integer interval, DexService service, TimeService timeService) {
+
+        int initialTime = toTs - (interval*limit);
+        int startGraph = initialTime;
+
+            
+        long startTS = (long)initialTime * 1000L;
+        long endTS = (long)toTs * 1000L; 
+            
+        Integer toTS = Convert2.toEpochTime(startTS);
+                                   
+        byte currencyType = 0;
+        
+        if ( tsym.equals("ETH") || fsym.equals("ETH") ) {
+            currencyType = 1;
+        } 
+                
+        if ( tsym.equals("PAX") || fsym.equals("PAX") ) {
+            currencyType = 0;
+        } 
+                                        
+        log.debug("start: {}, finish: {}, currencyType: {}, requestedType: {}", startTS, endTS, currencyType ); 
+                        
+        Integer startTSEpoch = Convert2.toEpochTime(startTS);
+        Integer endTSEpoch = Convert2.toEpochTime(endTS);
+            
+        log.debug("Epoch, start: {}, finish: {}", startTSEpoch, endTSEpoch ); 
+            
+        DexOrderDBRequestForTrading dexOrderDBRequestForTrading = new DexOrderDBRequestForTrading(startTSEpoch, endTSEpoch, (byte)1, currencyType, 0 , Integer.MAX_VALUE);
+            
+        List<DexOrder> dexOrdersForInterval = service.getOrdersForTrading(dexOrderDBRequestForTrading); 
+        
+        log.debug("found {} orders", dexOrdersForInterval.size() );
+        for (DexOrder cr : dexOrdersForInterval) {
+            log.trace("order: {}, amount: {}, a1: {}, a2: {}, rate: {},", cr.getId(), cr.getOrderAmount(), EthUtil.weiToEther(BigInteger.valueOf(cr.getOrderAmount())),
+                    EthUtil.fromAtm(BigDecimal.valueOf(cr.getOrderAmount())), cr.getPairRate());
+        }
+            
+        TradingDataOutput tradingDataOutput = new TradingDataOutput();            
+        tradingDataOutput.setResponse("Success");
+        tradingDataOutput.setType(100);
+        tradingDataOutput.setAggregated(false);
+            
+        List<SimpleTradingEntry> data = new ArrayList<>();
+            
+        BigDecimal prevClose= BigDecimal.ZERO;
+            
+        log.trace("extracted: {} values", dexOrdersForInterval.size() );
+            
+        for (int i=0; i< limit; i++) {                
+            long start = (long)initialTime * 1000L;
+            long finish = (interval * 1000L) + start ;                
+            Integer startEpoch =  Convert2.toEpochTime(start);
+            Integer finishEpoch =  Convert2.toEpochTime(finish);
+                
+            SimpleTradingEntry entryForPeriod = TradingViewUtils.getDataForPeriodFromOffersEpoch(dexOrdersForInterval, startEpoch, finishEpoch ); 
+            entryForPeriod.time = initialTime;
```

Comment:
You could use start variable to get the same effect. Initial time in such context lead to misunderstanding.